Fix h1 warning

sujinleeme opened the pull request:
This PR is to fix` h1` warnings in #181.
 
 Before https://rocketvalidator.com/s/41643d18-3a3f-4c34-85f7-fbdb1525636f/i?search=heading&page=1&non_doc=false

After changes
https://rocketvalidator.com/s/39d99f7a-b786-4c3b-a3d6-11783bdceb85/i?search=heading&page=0

jaimeiniesta commented on lib/school_house_web/templates/page/index.html.heex:
```diff
@@ -9,13 +9,13 @@
       </p>
       <p class="leading-relaxed">
-        Through the hard work of volunteers Elixir School has been translated to many languages. Some of these translations include: Việt ngữ, 简体中文, 繁體中文, Español, Slovenčina, 日本語, Polski, Português, Русский, Bahasa Indonesia, Bahasa Melayu, Українською, 한국어, Italiano, Deutsch, বাংলা, Türkçe, ภาษาไทย, and Ελληνικά.
+        Through the hard work of volunteers Elixir School has been translated to many languages. Some of thesef translations include: Việt ngữ, 简体中文, 繁體中文, Español, Slovenčina, 日本語, Polski, Português, Русский, Bahasa Indonesia, Bahasa Melayu, Українською, 한국어, Italiano, Deutsch, বাংলা, Türkçe, ภาษาไทย, and Ελληνικά.
```
There's a typo here, `thesef` should be `these`.

jaimeiniesta reviewed (changes requested):
I'm not sure about this, replacing the `section`s with `div`s does indeed remove the warnings, but I'm afraid we would be throwing the baby with the bath water here.

I think it makes sense to have the content on these pages organized in sections with their headings, I would try to keep the sections.

sujinleeme commented on lib/school_house_web/templates/page/index.html.heex:
```diff
@@ -244,9 +244,9 @@
   <div class="px-4 py-16 mx-auto text-center max-w-7xl sm:px-6 lg:py-16 lg:px-8">
-    <h2 class="text-3xl font-extrabold tracking-tight text-heavy dark:text-heavy-dark sm:text-4xl">
+    <h1 class="text-3xl font-extrabold tracking-tight text-heavy dark:text-heavy-dark sm:text-4xl">
       <span class="block">Want to learn more?</span>
```
@jaimeiniesta I think `<span class="block">` looks unnecessary.

sujinleeme commented on lib/school_house_web/templates/page/index.html.heex:
```diff
@@ -245,7 +245,7 @@
 <div class="bg-body dark:bg-body-dark">
```
Shouldn't be`<section>` ?

jaimeiniesta replied:
Our current design may have too many `div`s but to get a general idea, here's an explanation about `section` and headings:

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section

sujinleeme commented on lib/school_house_web/templates/page/index.html.heex:
```diff
@@ -244,9 +244,9 @@
-    <h2 class="text-3xl font-extrabold tracking-tight text-heavy dark:text-heavy-dark sm:text-4xl">
-      <span class="block">Want to learn more?</span>
-    </h2>
+    <h1 class="text-3xl font-extrabold tracking-tight text-heavy dark:text-heavy-dark sm:text-4xl">
+      Want to learn more?
+    </h1>
```
Then it could be `<h2>`.

jaimeiniesta replied:
Yes, I think so, the first one "Welcome to..." should be `h1` and the rest `h2`.

sujinleeme replied:
I changed `h2` to `h1` in here. e2fa6ad
`Welcome to part` is h1 in here, but `section` has to be changed to `div`  630a0b1.

jaimeiniesta commented:
Hi @sujinleeme - this does indeed remove the `h1` warnings but I think that heading structure in the home page is wrong, for example there are 2 `h1`s, one for the "Welcome" at the top and one for the "Recent posts" at the bottom. There should be only one `h1`, for the main section.

You can use the [HeadingsMap](https://chrome.google.com/webstore/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi/related) extension to get an outline of the headings structure, for example:

![Captura de pantalla 2021-11-02 a las 10 14 51](https://user-images.githubusercontent.com/2629/139818864-708c624f-4495-4a39-8942-38e1ba8151e9.png)

You can see this heading structure is a mess and doesn't correspond to the layout we see on the screen.

I think it would make more sense to have only these headings:

```
h1: Welcome to Elixir School!
  h2: Why Elixir?
  h2: By the Numbers
  h2: Recent Posts
```

Then, within "Why Elixir?", I would not use headings for the sub-sections "Functional Programming", "Feature Packed", as the content is so short. I think that a `p` with bold font style is enough. If headings are used, then it needs to be an `h3` because it's nested under an `h2`.

Lastly, the footer also uses headings but I think it shouldn't use them for the same reasons, but we can leave the footer for later.

sujinleeme commented:
@jaimeiniesta I had the same idea. That's a nice suggestion! 

![Screenshot 2021-11-02 at 13 07 31](https://user-images.githubusercontent.com/6011401/139843873-f6b94874-82eb-436b-a470-3a2b78329ed9.png)

I addressed your comment and I changed classes in `lib/school_house_web/templates/post/index.html.heex` 
https://github.com/elixirschool/school_house/pull/182/commits/cf25c8720ca6c7724abbe36d2cc3069126f1e3fb#

I will fix footer structure in different PR.

jaimeiniesta commented on lib/school_house_web/templates/page/index.html.heex:
```diff
@@ -1,4 +1,4 @@
-<section class="text-primary dark:text-primary-dark body-font">
+<div class="text-primary dark:text-primary-dark body-font">
```
I think it's now safe to make this a `section` again.

jaimeiniesta commented on lib/school_house_web/templates/page/index.html.heex:
```diff
@@ -35,7 +35,7 @@
     </div>
   </div>
-</section>
+</div>
```
Same here, this can be a `</section>`

sujinleeme replied:
I hope it would be fine now. :) df9c43c

jaimeiniesta replied:
That's great! 👍

jaimeiniesta reviewed (changes requested):
Cool, I've checked your branch, it looks good and fixes the issue.

I would bring back the `<section>` that contains the `<h1>`. It makes sense to define this as a section just like the other ones.

jaimeiniesta reviewed (approved):
I've checked it, this fixes the warnings and improves the heading structure. Good job @sujinleeme 🚀

sujinleeme commented:
@doomspork Could you tag`hacktoberfest` label? Thanks!
